### PR TITLE
feat(esl-carousel): add handler for 'esl-show-request'

### PR DESCRIPTION
**Pull request by NastaLeo**

### CODE REVIEW

File: src/modules/esl-carousel/core/esl-carousel.ts [hunk 1]
```diff
@@ -306,6 +306,15 @@ export class ESLCarousel extends ESLBaseElement {
     return el && el.hasAttribute('prev');
   }
 
+  // Navigates to requested slide
+  @listen('esl:show:request')
+  protected onHandleFootnotes(e: CustomEvent): void {
+    const target = (e.target as HTMLElement).closest('[esl-carousel-slide]') as HTMLElement;
```
Comment (ala-n) [inline src/modules/esl-carousel/core/esl-carousel.ts:312]: It looks good, but not perfect, just a couple of things we should consider here:

1) We have a really rare but realistic case when we have a nested carousel.
So in that case we will navigate trough the child carousel but we will not affect top parent carousel as the  `(e.target as HTMLElement).closest('[esl-carousel-slide]')` will give us only the most nested slide which is not member of parent carousel slide list.
We can discuss a solution for that case within the bounds of the current PR, or create that as a separate issue (If we really should harry up).

2) We had the support of the ` if (!isMatches(this, detail.match)) return null;` filter in toggleable, so we need to be consistent here as well.

File: src/modules/esl-carousel/core/esl-carousel.ts [hunk 2]
```diff
@@ -306,6 +306,15 @@ export class ESLCarousel extends ESLBaseElement {
...
+    const index = this.$slides.indexOf(target);
+    if (index === -1) return;
+    this.goTo(`slide:${index}`);
```
Comment (ala-n) [inline src/modules/esl-carousel/core/esl-carousel.ts:315]: Should be fine unless we have a bug
```suggestion
    if (index !== -1) this.goTo(index);
```

**Comment by ala-n** [comment]
Added `require squash` just in purpose to clarify public release note message a bit

File: src/modules/esl-carousel/core/esl-carousel.ts [hunk 3]
```diff
@@ -199,6 +205,14 @@ export class ESLCarousel extends ESLBaseElement {
     this.renderer && this.renderer.redraw();
   }
 
+  @listen('esl:show:request')
+  protected onHandleFootnotes(e: CustomEvent): void {
```
Comment (dshovchko) [inline src/modules/esl-carousel/core/esl-carousel.ts:209]: There is a confusion with the name
Probably should be

```suggestion
  protected onShowRequest(e: CustomEvent): void {
```

**Review by dshovchko — APPROVED** [review]

**Pull request merged**